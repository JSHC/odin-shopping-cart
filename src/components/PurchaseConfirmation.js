import { useEffect, useState } from "react";
import "../styles/PurchaseConfirmation.css";

const PurchaseConfirmation = (props) => {
    const [animate, setAnimate] = useState(false);

    useEffect(() => {
        setAnimate(true);
    });

    return (
        <div className={`purchase-confirmation ${animate ? "show" : "hidden"}`}>
            <h1>Thank you for your purchase!</h1>
            <button
                onClick={props.onClose}
                aria-label="close-purchase-confirmation"
            >
                CLOSE
            </button>
        </div>
    );
};

export default PurchaseConfirmation;
