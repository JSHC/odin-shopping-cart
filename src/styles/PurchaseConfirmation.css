.purchase-confirmation {
    position: absolute;
    z-index: 2;
    background-color: #fff;
    display: grid;
    justify-items: center;
    align-items: center;
    justify-self: center;
    align-self: center;
    padding: 4rem;
    box-shadow: rgba(100, 100, 111, 0.2) 0px 7px 29px 0px;
}

.show {
    transition: opacity 500ms;
    opacity: 1;
}

.hidden {
    opacity: 0;
}