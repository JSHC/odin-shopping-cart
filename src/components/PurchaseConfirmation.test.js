import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import PurchaseConfirmation from "./PurchaseConfirmation";
import { MemoryRouter } from "react-router-dom";
import userEvent from "@testing-library/user-event";

describe("PurchaseConfirmation component", () => {
    it("renders", () => {
        const { container } = render(
            <MemoryRouter>
                <PurchaseConfirmation />
            </MemoryRouter>
        );
        expect(container).toMatchSnapshot();
    });

    it("contains correct header text", () => {
        render(
            <MemoryRouter>
                <PurchaseConfirmation />
            </MemoryRouter>
        );
        screen.getByText("Thank you for your purchase!");
    });

    it("contains close button", () => {
        render(
            <MemoryRouter>
                <PurchaseConfirmation />
            </MemoryRouter>
        );
        const closeButton = screen.getByRole("button", {
            name: "close-purchase-confirmation",
        });
    });

    it("close button onClick works", async () => {
        const user = userEvent.setup();
        const mockCloseFn = jest.fn();
        render(
            <MemoryRouter>
                <PurchaseConfirmation onClose={mockCloseFn} />
            </MemoryRouter>
        );
        const closeButton = screen.getByRole("button", {
            name: "close-purchase-confirmation",
        });
        await user.click(closeButton);
        expect(mockCloseFn).toBeCalled();
    });
});
