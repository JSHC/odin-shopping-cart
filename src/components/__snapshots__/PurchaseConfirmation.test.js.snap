// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PurchaseConfirmation component renders 1`] = `
<div>
  <div
    class="purchase-confirmation show"
  >
    <h1>
      Thank you for your purchase!
    </h1>
    <button
      aria-label="close-purchase-confirmation"
    >
      CLOSE
    </button>
  </div>
</div>
`;
